DEV-8/src/Matrix.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Class matrix and methods to fill it and receive parameters
+ */
+public class Matrix {
```

Comment:
No need have two separate classes for Matrix and MatrixResults, use single one with BigDecimal matrix, If use double throw exception if overflow, it is normal